std/os/dragonfly.zig

```diff
@@ -0,0 +1,845 @@
+const builtin = @import("builtin");
+
+pub use @import("dragonfly/errno.zig");
```

Comment:
Same question as for `std/c/dragonfly.zig`: can this be shared with freebsd.zig? This is the only difference between those files, isn't it?

Reply:
I confess to having checked the other files for differences. But I left this one (other that errno). If there are differences with freebsd, I'm maybe not the person to fix them -- never having used 99% of what's in that file

Reply:
Originally yes it was copied. Now all defines are just wrappers around c/dragonfly.zig. Except for O_LARGEFILE. I initially special cased the dragonfly's absence of O_LARGEFILE in *.zig, but felt it was maybe better to just set O_LARGEFILE to the no-op of 0 for the moment